Plugins~/Src/MeshSyncClientBlender/msblenGeometryNodes.cpp

```diff
@@ -0,0 +1,167 @@
+    GeometryNodesUtils::GeometryNodesUtils()
+    {
+        auto rotation = rotate_x(-90 * DegToRad);
+        auto rotation180 = rotate_z(180 * DegToRad);
+        auto scale_z = float3::one();
+        scale_z.z = -1;
+
+        auto scale_x = float3::one();
+        scale_x.x = -1;
+
+        m_blender_to_unity_world =
+            to_mat4x4(rotation) *
+            scale44(scale_x);
+
+        m_blender_to_unity_local = 
+            to_mat4x4(rotation) *
+            to_mat4x4(rotation180) *
+            scale44(scale_z);
+    }
+
+    /// <summary>
+    /// Converts the world matrix from blender to Unity coordinate systems
+    /// </summary>
+    /// <param name="blenderMatrix"></param>
+    /// <returns></returns>
+    float4x4 GeometryNodesUtils::blenderToUnityWorldMatrix(const float4x4& blenderMatrix) {            
+
+        return 
+            m_blender_to_unity_world *
+            blenderMatrix *
+            m_blender_to_unity_local;
+    }
+
+    void GeometryNodesUtils::setInstancesDirty(bool dirty)
+    {
+        m_instances_dirty = dirty;
+    }
+    bool GeometryNodesUtils::getInstancesDirty()
+    {
+        return m_instances_dirty;
+    }
+
+    void GeometryNodesUtils::each_instanced_object(std::function<void(Object*, Object*, SharedVector<float4x4>, bool)> handler) {
+
+        std::unordered_map<unsigned int, Record> m_records;
+        std::unordered_map<std::string, Record*> m_records_by_name;
+
+        each_instance([&](Object* obj, Object* parent, float4x4 matrix)
+            {
+                // Critical path, must do as few things as possible
+                auto id = (ID*)obj->data;
+                auto& rec = m_records[id->session_uuid];
+                if (!rec.updated)
+                {
+                    rec.parent = parent;
+                     rec.object_copy = *obj;
+                    rec.updated = true;
+    
+                    m_records_by_name[obj->id.name] = &rec;
+                }
+
+                rec.matrices.push_back(matrix);
+            });
+
+            // Look for objects in the file
+            auto ctx = blender::BlenderPyContext::get();
+            auto objects = ctx.data()->objects;
+            LISTBASE_FOREACH(Object*, obj, &objects) {
+
+                if (obj->data == nullptr)
+                    continue;
+
+                // Check if there is record with the same object name
+                auto rec = m_records_by_name.find(obj->id.name);
+                if (rec == m_records_by_name.end())
+                    continue;
+
+                // Check if the data names also match
+                auto recDataId = (ID*)rec->second->object_copy.data;
+                auto sceneDataId = (ID*)obj->data;
+
+                if (strcmp(sceneDataId->name + 2, recDataId->name + 2) != 0)
+                    continue;
+
+                handler(obj, rec->second->parent, std::move(rec->second->matrices), true);
+                rec->second->handled = true;
```

Comment:
I used the wrong naming convention for records there. I used to keep them as private members of the class and clearing the maps at the end of the operation. I realized later its simpler to keep them as local variables in the function and refactored my code but did not rename them. 

So the records do not persist between executions of the function, it will give the same results. Sorry for the confusion with the names